### ISSUE TITLE
Type registration would not compile with clang on Windows

### ISSUE THREAD

**Issue by boguscoder** [USCiLab/cereal#582]
Minimal repro:
```
#include <cereal/types/polymorphic.hpp>

struct foo{
    virtual ~foo() = default;
};

CEREAL_REGISTER_TYPE_WITH_NAME(foo, "foo");

int main() {
    return 0;
}
```
Gets

> cereal/details/static_object.hpp:67:29: error:
>       'cereal::detail::StaticObject<cereal::detail::bind_to_archives<foo, cereal::detail::(anonymous
>       namespace)::polymorphic_binding_tag> >' must have external linkage when declared 'dllexport'
>     class CEREAL_DLL_EXPORT StaticObject
>                             ^
> foo.cpp:7:1: note: in instantiation of template class
>       'cereal::detail::StaticObject<cereal::detail::bind_to_archives<foo, cereal::detail::(anonymous
>       namespace)::polymorphic_binding_tag> >' requested here
> CEREAL_REGISTER_TYPE_WITH_NAME(foo, "foo");
> ^

This is caused by default polymorphic tag being in anonymous namespace, making it non-anon would fix the issue

---

EDIT: this is likely that https://bugs.llvm.org/show_bug.cgi?id=43471

**Comment by InBetweenNames** [comment]
Related: #595

**Comment by AzothAmmo** [comment]
reportedly fixed with #643
